src/main/java/com/google/step/YOUR_PROJECT_NAME_HERE/external/StackOverflowClient.java

```diff
@@ -0,0 +1,157 @@
+      HttpEntity entity = response.getEntity();
+      if (entity == null) {
+        return card;
+      }
+      BufferedReader reader = new BufferedReader(new InputStreamReader(entity.getContent()));
+      StringBuilder responseBody = new StringBuilder();
+      String line = null;
+      while ((line = reader.readLine()) != null) {
+        responseBody.append(line);
+      }
+      JSONObject res = new JSONObject(responseBody.toString());
+      String title = res.getJSONArray("items").getJSONObject(0).get("title").toString();
+      card.setLink(uri.toString());
+      // Store the id of the question in order to get the code body of the answer.
+      card.setCode(idStr);
+      card.setTitle(title);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    return card;
+  }
+
+  /* Get the most voted answer's id and store it in the card */
+  private Card getAnswerId(Card card) {
+    String url = String.format(QUESTION_URL_TEMPLATE, card.getCode());
+    CloseableHttpClient httpClient = HttpClients.createDefault();
+    try {
+      CloseableHttpResponse response = httpClient.execute(new HttpGet(url));
+      if (response.getStatusLine().getStatusCode() != 200) {
+        return card;
+      }
+      HttpEntity entity = response.getEntity();
+      if (entity == null) {
+        return card;
+      }
+      BufferedReader reader = new BufferedReader(new InputStreamReader(entity.getContent()));
+      StringBuilder responseBody = new StringBuilder();
+      String line = null;
+      while ((line = reader.readLine()) != null) {
+        responseBody.append(line);
+      }
+      JSONObject res = new JSONObject(responseBody.toString());
+      JSONArray answer_list = res.getJSONArray("items");
+      String answer_id = res.getJSONArray("items").getJSONObject(0).get("answer_id").toString();
+      // Replace the question id by the answer id in order to retrieve the code body next.
+      card.setCode(answer_id);
+    } catch (ClientProtocolException e) {
+      e.printStackTrace();
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    return card;
+  }
+
+  /* Get the content of the answer and store it in the card */
+  private Card getAnswer(Card card) {
+    String url = String.format(ANSWER_URL_TEMPLATE, card.getCode());
+    CloseableHttpClient httpClient = HttpClients.createDefault();
+    try {
+      CloseableHttpResponse response = httpClient.execute(new HttpGet(url));
+      if (response.getStatusLine().getStatusCode() != 200) {
+        return card;
+      }
+      HttpEntity entity = response.getEntity();
+      if (entity == null) {
+        return card;
+      }
+      BufferedReader reader = new BufferedReader(new InputStreamReader(entity.getContent()));
+      StringBuilder responseBody = new StringBuilder();
+      String line = null;
+      while ((line = reader.readLine()) != null) {
+        responseBody.append(line);
+      }
+      JSONObject res = new JSONObject(responseBody.toString());
```

Comment:
Most of the code here is repeated three times across all three helper functions. Is there a way you can minimize the repetition by moving this code into a separate helper function?